File: vacation_backend/Application/DTOs/Employee/EmployeeDetailedDto.cs
using vacation_backend.Application.DTOs.Vacation;
using vacation_backend.Domain.Enums;

namespace vacation_backend.Application.DTOs.Employee
{
    public class EmployeeDetailedDto
    {
        public int Id { get; set; }

        // Información personal
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public string FullName => $"{FirstName} {LastName}";
        public string? Email { get; set; }

        // Días de vacaciones
        public int AvailableDays { get; set; }
        public int UsedDays { get; set; }
        public int RemainingDays { get; set; }
        public bool HasAvailableDays { get; set; }

        // Beneficios extras
        public int RemainingExtraBenefitDays { get; set; }
        public bool HasAvailableExtraBenefitDays { get; set; }

        // Información laboral
        public int DepartmentId { get; set; }
        public string DepartmentName { get; set; } = null!;

        public int RoleId { get; set; }
        public string RoleName { get; set; } = null!;

        public EmployeeStatusEnum Status { get; set; }

        // Días extra asignados
        public List<EmployeeExtraBenefitDayDto>? ExtraBenefitDays { get; set; }

        // Solicitudes de vacaciones previas
        public List<VacationRequestHistoryDto>? VacationRequestHistory { get; set; }
    }
}

File: vacation-frontend/src/models/employee/EmployeeDetailedDto.ts
import type { EmployeeExtraBenefitDayDto } from './EmployeeExtraBenefitDayDto'
import type { Item } from '../ItemModel'
import { VacationRequestHistoryDto } from '../vacation/VacationRequestHistoryDto'

export interface EmployeeDetailedDto {
  id: number
  firstName: string
  lastName: string
  fullName: string
  email?: string | null

  availableDays: number
  usedDays: number
  remainingDays: number
  hasAvailableDays: boolean

  remainingExtraBenefitDays: number
  hasAvailableExtraBenefitDays: boolean

  departmentId: number
  departmentName: string

  roleId: number
  roleName: string

  status: Item

  extraBenefitDays?: EmployeeExtraBenefitDayDto[] | null
  vacationRequestHistory?: VacationRequestHistoryDto[] | null
}